File: Mathématiques/2 – Quelques outils mathématiques/08 - Champs circulaires.py
from math import *
def aire(rayon):
    return (rayon ** 2) * pi
def perim(rayon):
    return (rayon * 2) * pi

rayon_ext = 9

sum_perim = 0
sum_aire = 0

for i in range(5):
    rayon_int = rayon_ext - 3
    
    aire_disque = aire(rayon_ext) - aire(rayon_int)
    sum_aire += aire_disque
    
    perim_disque = perim(rayon_ext) + perim(rayon_int)
    sum_perim += perim_disque
    
    rayon_ext += 6

print(round(sum_aire))
print(round(sum_perim))

File: Mathématiques/2 – Quelques outils mathématiques/08 - Champs circulaires.cpp
#include <iostream>
#include <cmath>
using namespace std;

double perim(double rayon)
{
  return 2 * M_PI * rayon;
}
double aire(int rayon)
{
  return pow(rayon, 2) * M_PI;
}

int main()
{
  double sumAire = 0;
  double sumPerim = 0;
  int actualR = 3;
  for (int i = 0; i < 5; i++)
  {
    actualR += 3;
    double aireInterieur = aire(actualR);
    sumPerim += perim(actualR);
    actualR += 3;
    sumPerim += perim(actualR);
    double aireExterieur = aire(actualR);
    sumAire += aireExterieur - aireInterieur;
  }
  cout << round(sumAire) << endl
       << round(sumPerim);
}